pkg/list/list_stacks.go

```diff
@@ -1,43 +1,380 @@
+		var stacks []map[string]string
+		for _, row := range rows {
+			s := make(map[string]string)
+			for i, header := range headers {
+				// Only include non-empty values in JSON output
+				if row[i] != "" {
+					// Use raw value without any color formatting for JSON output
+					s[strings.ToLower(header)] = row[i]
+				}
+			}
+			stacks = append(stacks, s)
+		}
+		// Use plain JSON marshaling without any color formatting
+		jsonBytes, err := json.MarshalIndent(stacks, "", "  ")
+		if err != nil {
+			return "", fmt.Errorf("error formatting JSON output: %w", err)
+		}
+		return string(jsonBytes) + utils.GetLineEnding(), nil
+
+	case FormatCSV, FormatTSV:
+		// Only include columns that have values
+		var nonEmptyHeaders []string
+		var nonEmptyColumnIndexes []int
+
+		// Find columns that have at least one non-empty value
+		for i, header := range headers {
+			hasValue := false
+			for _, row := range rows {
+				if row[i] != "" {
+					hasValue = true
+					break
+				}
+			}
+			if hasValue {
+				nonEmptyHeaders = append(nonEmptyHeaders, header)
+				nonEmptyColumnIndexes = append(nonEmptyColumnIndexes, i)
+			}
+		}
+
+		// Set appropriate delimiter based on format
+		fileDelimiter := delimiter
+		if delimiter == "\t" || delimiter == "" {
+			switch format {
+			case FormatCSV:
+				fileDelimiter = ","
+			case FormatTSV:
+				fileDelimiter = "\t"
+			}
+		}
+
+		var output strings.Builder
+
+		switch format {
+		case FormatCSV:
+			// Use encoding/csv for proper CSV handling
+			writer := csv.NewWriter(&output)
+			writer.Comma = rune(fileDelimiter[0])
+
+			// Write headers
+			if err := writer.Write(nonEmptyHeaders); err != nil {
+				return "", fmt.Errorf("error writing CSV headers: %w", err)
+			}
+
+			// Write rows
+			for _, row := range rows {
+				csvRow := make([]string, len(nonEmptyColumnIndexes))
+				for j, i := range nonEmptyColumnIndexes {
+					csvRow[j] = row[i]
+				}
+				if err := writer.Write(csvRow); err != nil {
+					return "", fmt.Errorf("error writing CSV row: %w", err)
+				}
+			}
+			writer.Flush()
+			if err := writer.Error(); err != nil {
+				return "", fmt.Errorf("error flushing CSV writer: %w", err)
+			}
+
+		case FormatTSV:
+			// For TSV, replace tabs with spaces in values and use tab as delimiter
+			// Write headers
+			for i, header := range nonEmptyHeaders {
+				if i > 0 {
+					output.WriteString("\t")
+				}
+				output.WriteString(strings.ReplaceAll(header, "\t", " "))
+			}
+			output.WriteString(utils.GetLineEnding())
+
+			// Write rows
+			for _, row := range rows {
+				for j, i := range nonEmptyColumnIndexes {
+					if j > 0 {
+						output.WriteString("\t")
+					}
+					output.WriteString(strings.ReplaceAll(row[i], "\t", " "))
+				}
+				output.WriteString(utils.GetLineEnding())
+			}
+		}
+		return output.String(), nil
+
+	default:
+		// If format is empty or "table", use table format
+		if format == "" && term.IsTTYSupportForStdout() {
```

Comment:
My preference is we use the existing function in utils for checking for tty support, so we have one place to manage exceptions, such as forcing tty output.